src/verify_cert.rs

```diff
@@ -65,12 +59,15 @@ fn build_chain_inner(
     // for the purpose of name constraints checking, only end-entity server certificates
     // could plausibly have a DNS name as a subject commonName that could contribute to
     // path validity
-    let subject_common_name_contents =
-        if opts.required_eku_if_present == EKU_SERVER_AUTH && used_as_ca == UsedAsCa::No {
-            subject_name::SubjectCommonNameContents::DnsName
-        } else {
-            subject_name::SubjectCommonNameContents::Ignore
-        };
+    let eku = match opts.eku {
+        ExtendedKeyUsage::Required(eku) => eku,
+        ExtendedKeyUsage::RequiredIfPresent(eku) => eku,
+    };
```

Comment:
What do you think about making this `match` a `fn` in an `impl ExtendedKeyUsage`. Maybe something like `fn key_purpose(&self) -> KeyPurposeId`?

That would let you avoid repeating the same match here and on L388.